srv/components/misc_pkgs/kafka/stop.sls

```diff
@@ -25,24 +25,38 @@ Stop kafka:
   service.dead:
     - name: kafka
 
-#TODO: find better solution to add delay
-Wait for kafka to stop:
-  module.run:
-    - test.sleep:
-      - length: 10
+Ensure kafka has stopped:
+  cmd.run:
+    - name: test 1 -le $(ps ax | grep -i 'kafka.Kafka' | grep -v grep | awk '{print $1}' | wc -l)
```

Comment:
Use service command

Reply:
@sachinpunadikar the section on line 28 takes care of this requirement.
```
Stop kafka:
  service.dead:
    - name: kafka
 ```
Salt ensures that service has actually started in this block.